### DISCUSSION TITLE
Pyrebase

### DISCUSSION THREAD

**Discussion by merttalha** [flet-dev/flet#3241] [Q&A]
https://pypi.org/project/Pyrebase4/   A database library for making online applications and software

**Comment by FeodorFitsner** [comment]
It's a pure Python library. Will be working "as is" on any platform.

**Reply by merttalha** [reply]
PS C:\Python Flet\Game> & C:/Users/user/AppData/Local/Programs/Python/Python311/python.exe "c:/Python Flet/Game/main.py"
PS C:\Python Flet\Game> flet build apk                  
Creating Flutter bootstrap project...OK
Customizing app icons and splash images...OK
Generating app icons...OK   
Generating splash screens...OK
Packaging Python app...Running package command
Creating asset directory: C:\Users\user\AppData\Local\Temp\flet_flutter_build_mRrFSHRUdo\app
Copying Python app from C:\Python Flet\Game to C:\Users\user\AppData\Local\Temp\serious_python_temp4d86f2fd
Configured mobile platform with sitecustomize.py at C:\Users\user\AppData\Local\Temp\serious_python_sitecustomize26775ade\sitecustomize.py        
Installing dependencies  with pip command to C:\Users\user\AppData\Local\Temp\serious_python_temp4d86f2fd\__pypackages__
Extracting Python distributive from C:\Users\user\AppData\Local\Temp\cpython-3.11.6+20231002-x86_64-pc-windows-msvc-shared-install_only.tar.gz to 
C:\Users\user\AppData\Local\Temp\hostpython3.11_9f3a082a
Compiling Python sources at C:\Users\user\AppData\Local\Temp\serious_python_temp4d86f2fd
Error: FormatException: Invalid UTF-8 byte (at offset 169)
Deleting temp directory C:\Users\user\AppData\Local\Temp\serious_python_temp4d86f2fd

Unhandled exception:
FileSystemException: Deletion failed, path = 'C:\Users\user\AppData\Local\Temp\serious_python_temp4d86f2fd' (OS Error: Dizin boş değil.
, errno = 145)
#0      _checkForErrorResponse (dart:io/common.dart:55:9)
#1      _Directory._delete.<anonymous closure> (dart:io/directory_impl.dart:180:7)
<asynchronous suspension>
#2      PackageCommand.run (file:///C:/Users/user/AppData/Local/Pub/Cache/hosted/pub.dev/serious_python-0.7.0/bin/package_command.dart:354:9)
<asynchronous suspension>
#3      CommandRunner.runCommand (package:args/command_runner.dart:212:13)
<asynchronous suspension>
#4      main (file:///C:/Users/user/AppData/Local/Pub/Cache/hosted/pub.dev/serious_python-0.7.0/bin/main.dart:10:3)
<asynchronous suspension>

Error building Flet app - see the log of failed command above. Even so I get this error

**Reply by FeodorFitsner** [reply]
I doubt it's related to pyrebase package.

Please provide a full build log with `flet build apk -vv`.

**Reply by merttalha** [reply]
PS C:\Python Flet\Game> flet build apk -vv
Flutter executable: C:\src\flutter\bin\flutter.BAT
Dart executable: C:\src\flutter\bin\dart.BAT
Flutter bootstrap directory: C:\Users\user\AppData\Local\Temp\flet_flutter_build_6airuHBYzD
Additional Flutter dependencies:
{}
Creating Flutter bootstrap project...OK
Customizing app icons and splash images...Copying C:\Python Flet\Game\assets\icon.png to C:\Users\user\AppData\Local\Temp\flet_flutter_build_6airuHBYzD\images
OK
Generating app icons...
Run subprocess: ['C:\\src\\flutter\\bin\\dart.BAT', 'run', 'flutter_launcher_icons']
The Dart CLI developer tool uses Google Analytics to report usage and diagnostic
data along with package dependencies, and crash reporting to send basic crash
reports. This data is used to help improve the Dart platform, Flutter framework,
and related tools.

Telemetry is not sent on the very first run. To disable reporting of telemetry,
run this terminal command:

    dart --disable-analytics

If you opt out of telemetry, an opt-out event will be sent, and then no further
information will be sent. This data is collected in accordance with the Google
Privacy Policy (https://policies.google.com/privacy).

Resolving dependencies in C:\Users\user\AppData\Local\Temp\flet_flutter_build_6airuHBYzD... (1.7s)
Got dependencies in C:\Users\user\AppData\Local\Temp\flet_flutter_build_6airuHBYzD.
Building package executable... (2.6s)
Built flutter_launcher_icons:flutter_launcher_icons.
  ════════════════════════════════════════════
     FLUTTER LAUNCHER ICONS (v0.13.1)
  ════════════════════════════════════════════

• Creating default icons Android
• Overwriting the default Android launcher icon with a new icon
• Overwriting default iOS launcher icon with new icon
Creating Icons for Web...
Creating Icons for Windows...
Creating Icons for MacOS...

✓ Successfully generated launcher icons
OK
Generating splash screens...
Run subprocess: ['C:\\src\\flutter\\bin\\dart.BAT', 'run', 'flutter_native_splash:create']
Building package executable... (2.9s)
Built flutter_native_splash:create.
[Android] Creating default splash images
[Android] Creating dark mode splash images
[Android] Creating default android12splash images
[Android] Creating dark mode android12splash images
[Android] Updating launch background(s) with splash image path...
[Android]  - android/app/src/main/res/drawable/launch_background.xml
[Android]  - android/app/src/main/res/drawable-night/launch_background.xml
[Android]  - android/app/src/main/res/drawable-v21/launch_background.xml
[Android]  - android/app/src/main/res/drawable-night-v21/launch_background.xml
[Android] Updating styles...
[Android]  - android/app/src/main/res/values-v31/styles.xml
[Android] No android/app/src/main/res/values-v31/styles.xml found in your Android project
[Android] Creating android/app/src/main/res/values-v31/styles.xml and adding it to your Android project
[Android]  - android/app/src/main/res/values-night-v31/styles.xml
[Android] No android/app/src/main/res/values-night-v31/styles.xml found in your Android project
[Android] Creating android/app/src/main/res/values-night-v31/styles.xml and adding it to your Android project
[Android]  - android/app/src/main/res/values/styles.xml
[Android]  - android/app/src/main/res/values-night/styles.xml
[iOS] Creating  images
[iOS] Creating dark mode  images
[iOS] Updating ios/Runner/Info.plist for status bar hidden/visible
[Web] Creating images
[Web] Creating images
[Web] Creating background images
[Web] Creating CSS
[Web] Updating index.html
╔════════════════════════════════════════════════════════════════════════════╗
║                       NEED A GREAT FLUTTER DEVELOPER?                      ║
╠════════════════════════════════════════════════════════════════════════════╣
║                                                                            ║
║   I am available!  Find me at https://www.linkedin.com/in/hansonjon/       ║
║                                                                            ║
╚════════════════════════════════════════════════════════════════════════════╝

✅ Native splash complete.
Now go finish building something awesome! 💪 You rock! 🤘🤩
Like the package? Please give it a 👍 here: https://pub.dev/packages/flutter_native_splash

OK
Packaging Python app...
Run subprocess: ['C:\\src\\flutter\\bin\\dart.BAT', 'run', 'serious_python:main', 'package', 'C:\\Python Flet\\Game', '--mobile', '--platform', 'mobile', '--dep-mappings', 'flet=flet-embed',  
'--req-deps', 'flet-embed', '--exclude', 'build', '--verbose']
Building package executable... (1.5s)
Built serious_python:main.
Running package command
Creating asset directory: C:\Users\user\AppData\Local\Temp\flet_flutter_build_6airuHBYzD\app
Copying Python app from C:\Python Flet\Game to C:\Users\user\AppData\Local\Temp\serious_python_tempb8cf236c
Configured mobile platform with sitecustomize.py at C:\Users\user\AppData\Local\Temp\serious_python_sitecustomizefd1e209a\sitecustomize.py
Installing dependencies [flet-embed] with pip command to C:\Users\user\AppData\Local\Temp\serious_python_tempb8cf236c\__pypackages__
Extracting Python distributive from C:\Users\user\AppData\Local\Temp\cpython-3.11.6+20231002-x86_64-pc-windows-msvc-shared-install_only.tar.gz to C:\Users\user\AppData\Local\Temp\hostpython3.11_44c8fea1
VERBOSE: C:\Users\user\AppData\Local\Temp\hostpython3.11_44c8fea1\python\python.exe -m pip install --isolated --upgrade --target C:\Users\user\AppData\Local\Temp\serious_python_tempb8cf236c\__pypackages__ flet-embed
VERBOSE: Collecting flet-embed
VERBOSE: Obtaining dependency information for flet-embed from https://files.pythonhosted.org/packages/e8/b6/4044160cfc31a5e5fda4014bb2842fb1f1f2e3daf27ddb203a9b1ea8a2a9/flet_embed-0.22.0-py3-none-any.whl.metadata
VERBOSE: Using cached flet_embed-0.22.0-py3-none-any.whl.metadata (1.0 kB)
VERBOSE: Collecting flet-runtime==0.22.0 (from flet-embed)
VERBOSE: Obtaining dependency information for flet-runtime==0.22.0 from https://files.pythonhosted.org/packages/a6/12/5e4568476aaada6ea4f76eede18d15ba36bf3ac8108df061eabf1cee2604/flet_runtime-0.22.0-py3-none-any.whl.metadata
VERBOSE: Using cached flet_runtime-0.22.0-py3-none-any.whl.metadata (1.1 kB)
VERBOSE: Collecting flet-core==0.22.0 (from flet-runtime==0.22.0->flet-embed)
VERBOSE: Obtaining dependency information for flet-core==0.22.0 from https://files.pythonhosted.org/packages/f5/37/b54ccdd523281ca496d760f4b1d33970cddd8278d85e93b282d686b8bea0/flet_core-0.22.0-py3-none-any.whl.metadata
VERBOSE: Using cached flet_core-0.22.0-py3-none-any.whl.metadata (971 bytes)
VERBOSE: Collecting httpx<1,>=0 (from flet-runtime==0.22.0->flet-embed)
VERBOSE: Obtaining dependency information for httpx<1,>=0 from https://files.pythonhosted.org/packages/41/7b/ddacf6dcebb42466abd03f368782142baa82e08fc0c1f8eaa05b4bae87d5/httpx-0.27.0-py3-none-any.whl.metadata
VERBOSE: Using cached httpx-0.27.0-py3-none-any.whl.metadata (7.2 kB)
VERBOSE: Collecting oauthlib<4.0.0,>=3.2.2 (from flet-runtime==0.22.0->flet-embed)
VERBOSE: Obtaining dependency information for oauthlib<4.0.0,>=3.2.2 from https://files.pythonhosted.org/packages/7e/80/cab10959dc1faead58dc8384a781dfbf93cb4d33d50988f7a69f1b7c9bbe/oauthlib-3.2.2-py3-none-any.whl.metadata
VERBOSE: Using cached oauthlib-3.2.2-py3-none-any.whl.metadata (7.5 kB)
VERBOSE: Collecting repath<0.10.0,>=0.9.0 (from flet-core==0.22.0->flet-runtime==0.22.0->flet-embed)
VERBOSE: Obtaining dependency information for repath<0.10.0,>=0.9.0 from https://files.pythonhosted.org/packages/87/ed/92e9b8a3ffc562f21df14ef2538f54e911df29730e1f0d79130a4edc86e7/repath-0.9.0-py3-none-any.whl.metadata
VERBOSE: Using cached repath-0.9.0-py3-none-any.whl.metadata (899 bytes)
VERBOSE: Collecting anyio (from httpx<1,>=0->flet-runtime==0.22.0->flet-embed)
VERBOSE: Obtaining dependency information for anyio from https://files.pythonhosted.org/packages/14/fd/2f20c40b45e4fb4324834aea24bd4afdf1143390242c0b33774da0e2e34f/anyio-4.3.0-py3-none-any.whl.metadata
VERBOSE: Using cached anyio-4.3.0-py3-none-any.whl.metadata (4.6 kB)
VERBOSE: Collecting certifi (from httpx<1,>=0->flet-runtime==0.22.0->flet-embed)
VERBOSE: Obtaining dependency information for certifi from https://files.pythonhosted.org/packages/ba/06/a07f096c664aeb9f01624f858c3add0a4e913d6c96257acb4fce61e7de14/certifi-2024.2.2-py3-none-any.whl.metadata
VERBOSE: Using cached certifi-2024.2.2-py3-none-any.whl.metadata (2.2 kB)
VERBOSE: Collecting httpcore==1.* (from httpx<1,>=0->flet-runtime==0.22.0->flet-embed)
VERBOSE: Obtaining dependency information for httpcore==1.* from https://files.pythonhosted.org/packages/78/d4/e5d7e4f2174f8a4d63c8897d79eb8fe2503f7ecc03282fee1fa2719c2704/httpcore-1.0.5-py3-none-any.whl.metadata
VERBOSE: Using cached httpcore-1.0.5-py3-none-any.whl.metadata (20 kB)
VERBOSE: Collecting idna (from httpx<1,>=0->flet-runtime==0.22.0->flet-embed)
VERBOSE: Obtaining dependency information for idna from https://files.pythonhosted.org/packages/e5/3e/741d8c82801c347547f8a2a06aa57dbb1992be9e948df2ea0eda2c8b79e8/idna-3.7-py3-none-any.whl.metadata
VERBOSE: Using cached idna-3.7-py3-none-any.whl.metadata (9.9 kB)
VERBOSE: Collecting sniffio (from httpx<1,>=0->flet-runtime==0.22.0->flet-embed)
VERBOSE: Obtaining dependency information for sniffio from https://files.pythonhosted.org/packages/e9/44/75a9c9421471a6c4805dbf2356f7c181a29c1879239abab1ea2cc8f38b40/sniffio-1.3.1-py3-none-any.whl.metadata
VERBOSE: Using cached sniffio-1.3.1-py3-none-any.whl.metadata (3.9 kB)
VERBOSE: Collecting h11<0.15,>=0.13 (from httpcore==1.*->httpx<1,>=0->flet-runtime==0.22.0->flet-embed)
VERBOSE: Obtaining dependency information for h11<0.15,>=0.13 from https://files.pythonhosted.org/packages/95/04/ff642e65ad6b90db43e668d70ffb6736436c7ce41fcc549f4e9472234127/h11-0.14.0-py3-none-any.whl.metadata
VERBOSE: Using cached h11-0.14.0-py3-none-any.whl.metadata (8.2 kB)
VERBOSE: Collecting six>=1.9.0 (from repath<0.10.0,>=0.9.0->flet-core==0.22.0->flet-runtime==0.22.0->flet-embed)
VERBOSE: Obtaining dependency information for six>=1.9.0 from https://files.pythonhosted.org/packages/d9/5a/e7c31adbe875f2abbb91bd84cf2dc52d792b5a01506781dbcf25c91daf11/six-1.16.0-py2.py3-none-any.whl.metadata
VERBOSE: Using cached six-1.16.0-py2.py3-none-any.whl.metadata (1.8 kB)
VERBOSE: Using cached flet_embed-0.22.0-py3-none-any.whl (3.0 kB)
VERBOSE: Using cached flet_runtime-0.22.0-py3-none-any.whl (19 kB)
VERBOSE: Using cached flet_core-0.22.0-py3-none-any.whl (366 kB)
VERBOSE: Using cached httpx-0.27.0-py3-none-any.whl (75 kB)
VERBOSE: Using cached httpcore-1.0.5-py3-none-any.whl (77 kB)
VERBOSE: Using cached oauthlib-3.2.2-py3-none-any.whl (151 kB)
VERBOSE: Using cached repath-0.9.0-py3-none-any.whl (4.7 kB)
VERBOSE: Using cached anyio-4.3.0-py3-none-any.whl (85 kB)
VERBOSE: Using cached idna-3.7-py3-none-any.whl (66 kB)
VERBOSE: Using cached sniffio-1.3.1-py3-none-any.whl (10 kB)
VERBOSE: Using cached certifi-2024.2.2-py3-none-any.whl (163 kB)
VERBOSE: Using cached h11-0.14.0-py3-none-any.whl (58 kB)
VERBOSE: Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
VERBOSE: Installing collected packages: sniffio, six, oauthlib, idna, h11, certifi, repath, httpcore, anyio, httpx, flet-core, flet-runtime, flet-embed
VERBOSE: Successfully installed anyio-4.3.0 certifi-2024.2.2 flet-core-0.22.0 flet-embed-0.22.0 flet-runtime-0.22.0 h11-0.14.0 httpcore-1.0.5 httpx-0.27.0 idna-3.7 oauthlib-3.2.2 repath-0.9.0 
six-1.16.0 sniffio-1.3.1
Compiling Python sources at C:\Users\user\AppData\Local\Temp\serious_python_tempb8cf236c
VERBOSE: C:\Users\user\AppData\Local\Temp\hostpython3.11_44c8fea1\python\python.exe -m compileall -b C:\Users\user\AppData\Local\Temp\serious_python_tempb8cf236c
Error: FormatException: Invalid UTF-8 byte (at offset 169)
Deleting temp directory C:\Users\user\AppData\Local\Temp\serious_python_tempb8cf236c
Unhandled exception:
FileSystemException: Deletion failed, path = 'C:\Users\user\AppData\Local\Temp\serious_python_tempb8cf236c' (OS Error: Dizin boş değil.
, errno = 145)
#0      _checkForErrorResponse (dart:io/common.dart:55:9)
#1      _Directory._delete.<anonymous closure> (dart:io/directory_impl.dart:180:7)
<asynchronous suspension>
#2      PackageCommand.run (file:///C:/Users/user/AppData/Local/Pub/Cache/hosted/pub.dev/serious_python-0.7.0/bin/package_command.dart:354:9)
<asynchronous suspension>
#3      CommandRunner.runCommand (package:args/command_runner.dart:212:13)
<asynchronous suspension>
#4      main (file:///C:/Users/user/AppData/Local/Pub/Cache/hosted/pub.dev/serious_python-0.7.0/bin/main.dart:10:3)
<asynchronous suspension>
Deleting Flutter bootstrap directory C:\Users\user\AppData\Local\Temp\flet_flutter_build_6airuHBYzD
Error building Flet app - see the log of failed command above.

**Reply by FeodorFitsner** [reply]
See if you still have `C:\Users\user\AppData\Local\Temp\serious_python_tempb8cf236c` directory (or CTRL+C before that error next time) and try compiling Python code there with `python.exe -m compileall -b` (get full command from the log).

**Reply by FeodorFitsner** [reply]
Looks like you just posted a code with an api key. I've deleted the post. Regenerate the key immediately and invalidate the old one. Be careful while posting secrets online.

**Reply by merttalha** [reply]
import flet as ft
import pyrebase  # Firebase kütüphanesini ekledik

# 03.05.2024
# Firebase yapılandırma bilgileri
config = {
  "apiKey": "",
  "authDomain": "",
  "projectId": "",
  "storageBucket": "",
  "messagingSenderId": "",
  "appId": "",
  "databaseURL": "",
}

firebase = pyrebase.initialize_app(config)
auth = firebase.auth()

# Input field style
input_style = {
    "height": 38,
    "focused_border_color": "#55a271",
    "border_radius": 5,
    "cursor_height": 16,
    "cursor_color": "#55a271",
    "content_padding": 10,
    "border_width": 1.5,
    "text_size": 12,
}

class Input(ft.TextField):
    def __init__(self, password: bool):
        super().__init__(**input_style, password=password)

# Button style
button_style = {
    "expand": True,
    "height": 38,
    "bgcolor": "#55a271",
    "style": ft.ButtonStyle(shape={"": ft.RoundedRectangleBorder(radius=5)}),
    "color": "white",
}

class Button(ft.ElevatedButton):
    def __init__(self, text: str, function: callable):
        super().__init__(
            **button_style,
            text=text,
            on_click=function,
        )

# Body style
body_style = {
    "width": 400,
    "height": 420,
    "border_radius": 8,
    "padding": 10,
}

class Body(ft.Container):
    def __init__(
        self,
        title: str,
        btn_name: str,
        msg1: str,
        msg2: str,
        route: str,
        function: callable,
    ):
        super().__init__(**body_style)
        self.email = Input(password=False)
        self.password = Input(password=True)

        self.content = ft.Column(
            controls=[
                ft.Text(title, size=21, color="#55a271"),
                ft.Column(width=400, controls=[ft.Divider(height=5, color="white")]),
                ft.Text("Email", size=10),
                self.email,
                ft.Divider(height=5, color="transparent"),
                ft.Text("Password", size=10),
                self.password,
                ft.Divider(height=10, color="transparent"),
                ft.Row(controls=[Button(btn_name, function)]),
                ft.Divider(height=15, color="transparent"),
                ft.Row(
                    alignment="center",
                    controls=[
                        ft.Text(
                            msg1,
                            size=12,
                            opacity=0.7,
                            spans=[
                                ft.TextSpan(
                                    msg2,
                                    style=ft.TextStyle(weight="bold"),
                                    on_click=lambda _: self.page.go(route),
                                )
                            ],
                        )
                    ],
                ),
            ]
        )

class LogInPage(ft.View):
    def __init__(self, page: ft.Page):
        super().__init__(
            route="/login-user",
            vertical_alignment="center",
            horizontal_alignment="center",
        )
        self.page = page
        self.body = Body(
            title="Giriş Yap",
            btn_name="Giriş Yap",
            msg1="Hesabınız yok mu? Burada bir tane ",
            msg2="oluşturun",
            route="/create-user",
            function=self.log_user_in,
        )
        self.controls = [self.body]

    def log_user_in(self, event=None):
        try:
            user = auth.sign_in_with_email_and_password(
                self.body.email.value,  # Email'i al
                self.body.password.value,  # Şifreyi al
            )

            # Kullanıcı kimliğini sakla
            self.page.user_id = user['localId']

            # Kullanıcı kimliğini dosyaya yaz
            with open("C:/Python Flet/Game/assets/Auth/ID.txt", "w") as file:
                file.write(self.page.user_id)

            # Giriş başarılı olduğunda yapılacak işlemler
            print("Giriş yapıldı!")
            self.page.go("/home")  # Anasayfaya yönlendir
        except Exception as e:
            # Hata durumunda
            print("Giriş yapılamadı:", e)

class CreatePage(ft.View):
    def __init__(self, page: ft.Page):
        super().__init__(
            route="/create-user",
            vertical_alignment="center",
            horizontal_alignment="center",
        )
        self.page = page
        self.body = Body(
            title="Kayıt Ol",
            btn_name="Hesap Oluştur",
            msg1="Hesabın var mı? ",
            msg2="Giriş yap",
            route="/login-user",
            function=self.create_new_user,
        )
        self.controls = [self.body]

    def create_new_user(self, event=None):
        try:
            user = auth.create_user_with_email_and_password(
                self.body.email.value,  # Email'i al
                self.body.password.value,  # Şifreyi al
            )

            # Kullanıcı kimliğini sakla
            self.page.user_id = user['localId']

            # Kullanıcı kimliğini dosyaya yaz
            with open("C:/Python Flet/Game/assets/Auth/ID.txt", "w") as file:
                file.write(self.page.user_id)

            # Kullanıcı oluşturma başarılı olduğunda yapılacak işlemler
            print("Kullanıcı oluşturuldu!")
            self.page.go("/home")  # Anasayfaya yönlendir
        except Exception as e:
            # Hata durumunda
            print("Kullanıcı oluşturulamadı:", e)

class HomePage(ft.View):
    global menu_index
    def __init__(self, page: ft.Page):
        super().__init__(
            route="/home",
            vertical_alignment="",
            horizontal_alignment="center",
            scroll=True
        )
        self.page = page
        menu = page.navigation_bar = ft.CupertinoNavigationBar(
            bgcolor="#55a271",
            inactive_color=ft.colors.WHITE,
            active_color=ft.colors.BLACK,
            on_change=lambda e: print("Selected tab:", e.control.selected_index),
            destinations=[
                ft.NavigationDestination(icon=ft.icons.HOME, label="Ana Sayfa"),
                ft.NavigationDestination(icon=ft.icons.SEARCH, label="Ara"),
                ft.NavigationDestination(icon=ft.icons.CHAT, label="Sohbetler"),
                ft.NavigationDestination(icon=ft.icons.PERSON, label="Profilim"),

            ]
        )
        # Post Shcema 
        post = ft.Column(
            alignment="center",
            controls=[
                ft.Row(
                    controls=[
                        ft.Column(
                            width=0,
                            controls=[
                                ft.Divider(color="white",height=6)
                            ]
                        )
                    ]
                ),
                ft.Row(
                    controls=[
                        ft.Image(src="./assets/kodular.png",height=30,width=30),
                        ft.Text(value="calismert3")
                    ]
                ),
                ft.Row(
                    controls=[
                        ft.Column(
                            width=400,
                            controls=[
                                ft.Divider(color="white",height=0)
                            ]
                        )
                    ]
                ),
                ft.Row(
                    controls=[
                        ft.Image(src="./assets/post.png",height=400,width=400)
                    ]
                ),
                ft.Row(
                    controls=[
                        ft.Column(
                            width=400,
                            controls=[
                                ft.Divider(color="white",height=0)
                            ]
                        )
                    ]
                ),
                ft.Row(
                    controls=[
                        ft.ElevatedButton(
                            bgcolor="white",
                            width=90,
                            content=ft.Container(
                                content=ft.Column(
                                    [
                                        ft.Image(src="./assets/heartof.png",height=30,width=30)
                                    ],
                                    alignment=ft.MainAxisAlignment.CENTER,
                                    spacing=5,
                                ),
                                padding=ft.padding.all(10),
                            )
                        ),
                        ft.ElevatedButton(
                            bgcolor="white",
                            width=90,
                            content=ft.Container(
                                content=ft.Column(
                                    [
                                        ft.Image(src="./assets/comment.png",height=30,width=30)
                                    ],
                                    alignment=ft.MainAxisAlignment.CENTER,
                                    spacing=5,
                                ),
                                padding=ft.padding.all(10),
                            )
                        )
                    ]
                ),
                
            ]
        )

        # Sayfa içeriği
        self.controls = [
            ft.AppBar(title=ft.Text('Ana Sayfa'), bgcolor='#55a271'),
            ft.Column(
                
                alignment="center",
                controls=[
                    ft.Row(
                        alignment="center",
                        controls=[
                            ft.TextField(height=38,width=350,focused_border_color="#55a271",border_radius=5,cursor_height=16,cursor_color="#55a271",content_padding=10,border_width=1.5,text_size=12,color="#55a271",hint_text="Ara..."),
                            ft.IconButton(icon=ft.icons.SEARCH,icon_color="#55a271", on_click=None,),   
                        ]   
                    ),
                    ft.Row(
                        alignment="center",
                        controls=[
                            post
                        ],                    
                    ),
                    ft.Row(
                        alignment="center",
                        controls=[
                            post
                        ],                    
                    ),
                    ft.Row(
                        alignment="center",
                        controls=[
                            post
                        ],                    
                    ),
                    ft.Row(
                        alignment="center",
                        controls=[
                            post
                        ],                    
                    ),
                    ft.Row(
                        alignment="center",
                        controls=[
                            post
                        ],                    
                    ),
                ]
                
            ),
            menu
            

            
        ]

def main(page: ft.Page):
    theme = ft.Theme()
    theme.page_transitions.windows = ft.PageTransitionTheme.NONE
    page.theme = theme

    page.bgcolor = "#202020"

    create: ft.View = CreatePage(page)
    login: ft.View = LogInPage(page)
    home: ft.View = HomePage(page)  # Anasayfa ekleniyor

    # Kullanıcı kimliğini saklamak için bir değişken ekle
    page.user_id = None

    # Uygulama başladığında dosyadan kullanıcı kimliğini kontrol et
    try:
        with open("C:/Python Flet/Game/assets/Auth/ID.txt", "r") as file:
            page.user_id = file.read().strip()
    except FileNotFoundError:
        pass

    # Yönlendirici yöntemi
    def route_change(route):
        page.views.clear()

        if page.route == "/create-user":
            page.views.append(create)
        if page.route == "/login-user":
            page.views.append(login)
        if page.route == "/home":  # Anasayfaya yönlendirme eklendi
            page.views.append(home)

        # Uygulama başladığında kullanıcı kimliğini kontrol et
        if page.user_id:
            page.go("/home")

        page.update()

    page.on_route_change = route_change
    page.go("/create-user")

ft.app(target=main, assets_dir="assets")



I couldn't find a solution. Could you please review my code?

**Reply by FeodorFitsner** [reply]
I'm sorry I can't help with code review.

Instead, I'm asking you to try compiling your Python app code (make a copy of it before doing that!) with `python.exe -m compileall -b` command and see if it produces any errors. Could be the encoding of some file or anything.

**Reply by merttalha** [reply]
A file named main.pyc was created

**Reply by merttalha** [reply]
Traceback (most recent call last):
  File "<string>", line 43, in <module>
  File "<frozen runpy>", line 229, in run_module
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\user\AppData\Local\Temp\serious_python_temp8abda71a\main.py", line 2, in <module>
ModuleNotFoundError: No module named 'pyrebase'


I print out the APK and see it on my app screen
